components/board/pi/impl/board.go

```diff
@@ -601,7 +605,7 @@ func pigpioInterruptCallback(gpio, level int, rawTick uint32) {
 		}
 		// this should *not* block for long otherwise the lock
 		// will be held
-		err := i.Tick(context.TODO(), high, tick*1000)
+		err := i.Tick(instanceCtx, high, tick)
```

Comment:
We used to have a new context for each function call, and now we reuse the same one for all of them. What prompted this change? I suspect either the old way or the new way is fundamentally unsafe, but I can't tell which.

Reply:
it's so that the ticks are cancelled when the board is closed

Reply:
Would there ever be a case when there are two interrupts and you want to cancel one but not the other? Like, maybe you've got an ultrasonic sensor hooked up to your board, and you also have a motion sensor that sends an interrupt if it's in freefall (which we don't yet support but I'd like to eventually) or something like that. If this question turns out to be hard to answer or the code change gets thorny, my preference is to undo the change to this line and make a separate PR to address it instead. The rest of this PR is too cool to be held up over this, and two small PRs are easier to review than one big one.

Reply:
It looks like you've changed this to have a new function for each interrupt, rather than one global one. Thanks!